dask/dataframe/core.py

```diff
@@ -7775,7 +7782,7 @@ def to_datetime(arg, meta=None, **kwargs):
 def to_timedelta(arg, unit=None, errors="raise"):
     if not PANDAS_GT_110 and unit is None:
         unit = "ns"
-    meta = pd.Series([pd.Timedelta(1, unit=unit)])
+    meta = meta_series_constructor(arg)([pd.Timedelta(1, unit=unit)])
```

Comment:
(Not a blocking comment, just meant for other reviewers) Noting that this will fail for some inputs (e.g. `numpy.ndarray`s) that are supported by `pandas.to_timedelta`. However, it looks like things already fail with `dd.to_timedelta` for such inputs on `main` and we arguably get a better error message with the changes in this PR. 

On `main`:

```python
In [1]: import numpy as np

In [2]: import dask.dataframe as dd

In [3]: dd.to_timedelta(np.arange(5), unit='s')
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
Input In [3], in <cell line: 1>()
----> 1 dd.to_timedelta(np.arange(5), unit='s')

File ~/projects/dask/dask/dask/dataframe/core.py:7779, in to_timedelta(arg, unit, errors)
   7777     unit = "ns"
   7778 meta = pd.Series([pd.Timedelta(1, unit=unit)])
-> 7779 return map_partitions(pd.to_timedelta, arg, unit=unit, errors=errors, meta=meta)

File ~/projects/dask/dask/dask/dataframe/core.py:6668, in map_partitions(func, meta, enforce_metadata, transform_divisions, align_dataframes, *args, **kwargs)
   6665     if collections:
   6666         simple = False
-> 6668 divisions = _get_divisions_map_partitions(
   6669     align_dataframes, transform_divisions, dfs, func, args, kwargs
   6670 )
   6672 if has_keyword(func, "partition_info"):
   6673     partition_info = {
   6674         (i,): {"number": i, "division": division}
   6675         for i, division in enumerate(divisions[:-1])
   6676     }

File ~/projects/dask/dask/dask/dataframe/core.py:6711, in _get_divisions_map_partitions(align_dataframes, transform_divisions, dfs, func, args, kwargs)
   6707 """
   6708 Helper to get divisions for map_partitions and map_overlap output.
   6709 """
   6710 if align_dataframes:
-> 6711     divisions = dfs[0].divisions
   6712 else:
   6713     # Unaligned, dfs is a mix of 1 partition and 1+ partition dataframes,
   6714     # use longest divisions found
   6715     divisions = max((d.divisions for d in dfs), key=len)

IndexError: list index out of range
``` 

With this PR:

```python
In [4]: dd.to_timedelta(np.arange(5), unit='s')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [4], in <cell line: 1>()
----> 1 dd.to_timedelta(np.arange(5), unit='s')

File ~/projects/dask/dask/dask/dataframe/core.py:7785, in to_timedelta(arg, unit, errors)
   7783 if not PANDAS_GT_110 and unit is None:
   7784     unit = "ns"
-> 7785 meta = meta_series_constructor(arg)([pd.Timedelta(1, unit=unit)])
   7786 return map_partitions(pd.to_timedelta, arg, unit=unit, errors=errors, meta=meta)

File ~/projects/dask/dask/dask/dataframe/utils.py:782, in meta_series_constructor(like)
    780     return like.to_frame()._constructor_sliced
    781 else:
--> 782     raise TypeError(f"{type(like)} not supported by meta_series_constructor")

TypeError: <class 'numpy.ndarray'> not supported by meta_series_constructor
```